doc/branches.md

```diff
@@ -1 +1,1719 @@
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+404 | Not Found | User not found in database  
+500 | Internal Server Error | some un-handle error in server 
+204 | No Content | Server request has succeeded but response is no content
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "Not Found",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-204",
+    "status_message": "Data Has Been Deleted",
+    "data": null
+}
+```
+
+### Logic
+
+#### Endpoint validation
+must be add parameter id and must exist in database
+
+#### Headers Validation
+- email : user email required and is not empty
+- email : check user email is login or not
+- email : must be top level, like administrator
+
+#### Request Payloads Validation
+No Request Payloads Validation
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Headers : empty
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "headers email is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Headers : email is not login
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "headers email is not login",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Headers : email must be top level access
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "headers email must be top level access or administrator",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+- param id is empty
+
+Endpoint : /branches
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "url Not Found, param id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+- param id is not exist in database
+
+Endpoint : /branches/359ed520-346f-11eb-adc1-0242ac120112
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "data not found",
+    "data": null
+}
+```
+
+#### Case : Positive Case
+
+- param id is exist in database
+
+Endpoint : /branches/632c725e-35e6-11eb-adc1-0242ac120002
+
+Headers :
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@gmail.com
+
+Response HTTP Status Code : 204
+
+Response Payload :
+```
+{
+    "status_code": "CDC-204",
+    "status_message": "Data Has Been Deleted",
+    "data": null
+}
+```
+
+## <a name="get-all-branches"></a>Get All
+
+### Endpoint
+GET /branches
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@gmail.com
+
+### Request Payloads
+No Request Payloads
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+404 | Not Found | User not found in database  
```

Comment:
for list, prefer there is no 404. you should use 200 status code with the empty data.